src/signalrclient/connection_impl.cpp

```diff
@@ -397,13 +442,188 @@ namespace signalr
+            m_start_completed_event.reset();
+        }
+
+        try
+        {
+            m_reconnecting();
+        }
+        catch (const std::exception &e)
+        {
+            m_logger.log(
+                trace_level::errors,
+                utility::string_t(_XPLATSTR("reconnecting callback threw an exception: "))
+                .append(utility::conversions::to_string_t(e.what())));
+        }
+        catch (...)
+        {
+            m_logger.log(
+                trace_level::errors,
+                utility::string_t(_XPLATSTR("reconnecting callback threw an unknown exception")));
+        }
+
+        auto reconnect_url = url_builder::build_reconnect(m_base_url, m_transport->get_transport_type(),
+            m_connection_token, m_connection_data, m_message_id, m_groups_token, m_query_string);
+
+        auto weak_connection = std::weak_ptr<connection_impl>(shared_from_this());
+
+        // this is non-blocking
+        try_reconnect(reconnect_url, utility::datetime::utc_now().to_interval(), m_reconnect_window, m_reconnect_delay, m_disconnect_cts)
+            .then([weak_connection](pplx::task<bool> reconnect_task)
+            {
+                // try reconnect does not throw
+                auto reconnected = reconnect_task.get();
+
+                auto connection = weak_connection.lock();
+                if (!connection)
+                {
+                    // connection instance went away - nothing to be done
+                    return pplx::task_from_result();
+                }
+
+                connection->m_start_completed_event.set();
+
+                if (reconnected)
+                {
+                    if (!connection->change_state(connection_state::reconnecting, connection_state::connected))
+                    {
+                        connection->m_logger.log(trace_level::errors,
+                            utility::string_t(_XPLATSTR("internal error - transition from an unexpected state. expected state: reconnecting, actual state: "))
+                            .append(translate_connection_state(connection->get_connection_state())));
+
+                        _ASSERTE(false);
+                    }
+
+                    try
+                    {
+                        connection->m_reconnected();
+                    }
+                    catch (const std::exception &e)
+                    {
+                        connection->m_logger.log(
+                            trace_level::errors,
+                            utility::string_t(_XPLATSTR("reconnected callback threw an exception: "))
+                            .append(utility::conversions::to_string_t(e.what())));
+                    }
+                    catch (...)
+                    {
+                        connection->m_logger.log(
+                            trace_level::errors,
+                            utility::string_t(_XPLATSTR("reconnected callback threw an unknown exception")));

```

Comment:
nit: You removed the explicit constructor call to utility::string_t earlier in this PR but have it here